libmattermost.c

```diff
@@ -3700,22 +3702,19 @@ if (len == MATTERMOST_HISTORY_PAGE_SIZE && channel->page_history < MATTERMOST_MA
 		mm_g_free_mattermost_channel(channel);
 	}
 // for now we could just tell user...
-
 }
 
 static void
-mm_get_history_of_room(MattermostAccount *ma, MattermostChannel *channel, gint64 since)
+mm_get_history_of_room(MattermostAccount *ma, MattermostChannel *channel)
 {
 	gchar *url;
 
-	if (channel->page_history == MATTERMOST_MAX_PAGES) return;
+	// case for recursion to end
+	if (channel->page_history < 0) return;
 	if (!channel->id) return;
 
-	if (since < 0) {
-		since = mm_get_channel_approximate_view_time(ma, channel);
-	}
-
-	url = mm_build_url(ma,"/channels/%s/posts?page=%s&per_page=%s&since=%" G_GINT64_FORMAT "", channel->id, g_strdup_printf("%i",channel->page_history), g_strdup_printf("%i", MATTERMOST_HISTORY_PAGE_SIZE), since);
+	// api call to fetch previous posts with page and per_page parameter
+	url = mm_build_url(ma,"/channels/%s/posts?page=%s&per_page=%s", channel->id, g_strdup_printf("%i",channel->page_history), g_strdup_printf("%i", MATTERMOST_HISTORY_PAGE_SIZE));

```

Comment:
We should define a string constant for this URL `/channels/%s/posts?page=%s&per_page=%s`

Reply:
constants are not maintained for any URLs (38 are there)
if we have to modify this then it will need to be done throughout the project for consistency, so, we can do it as a refactoring part later.